viewing_party/party.py

```diff
@@ -0,0 +1,141 @@
+def create_movie(movie_title, genre, rating): 
+    if movie_title == None \
+    or genre == None \
+    or rating == None:  
+        return None
+    
+    return {
+        "title": movie_title, 
+        "genre": genre, 
+        "rating": rating
+    }
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+
+    return user_data
+
+def add_to_watchlist(user_data, movie): 
+    user_data["watchlist"].append(movie)
+
+    return user_data
+
+def watch_movie(user_data, title): 
+    watch_list = user_data["watchlist"]
+    
+    for movie in watch_list: 
+        if title == movie.get("title"):
+            watch_list.remove(movie)
+            updated_data = add_to_watched(user_data, movie)
+            return updated_data
+
+    return user_data
+
+def get_watched_avg_rating(user_data):
+    watched = user_data["watched"]
+
+    if not user_data["watched"]: 
+        return 0
+
+    sum = 0
+    for movie in watched: 
+        movie_rating = movie.get("rating")
+        sum += movie_rating
+    
+    if sum == 0: 
+        return 0
+    else: 
+        avg_rating = sum/len(user_data["watched"])
+        return avg_rating
+
+def get_most_watched_genre(user_data):
+    watched = user_data["watched"]
+    most_watched_genre = {}
+    
+    for movie in watched: 
+        genre = movie.get("genre")
+        if genre in most_watched_genre: 
+            most_watched_genre[genre] += 1
+        else: 
+            most_watched_genre[genre] = 0
+    
+    if not most_watched_genre: 
+        return None
+
+    most_watched_genre = \
+        max(most_watched_genre, key = most_watched_genre.get)  
+    
+    return most_watched_genre
+
+def get_friends_watched(user_data):
+    friends_data = user_data["friends"]
+    friends_watched_list = []
+
+    for friend in friends_data:
+        for movie in friend["watched"]:
+            if movie in friends_watched_list: 
+                continue
```

Comment:
Just noting you could also use a set here.

Reply:
yes! I initially started the comparing test waves using all sets, but then couldn't figure out how to turn those back into lists, so I switched to loops. Excited to implement sets in future projects though. Thank you for reminding me of this!